internal/provider/inventory_data_source.go

```diff
@@ -4,103 +4,97 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
-	"net/http"
-	"slices"
 
+	"github.com/hashicorp/terraform-plugin-framework-jsontypes/jsontypes"
 	"github.com/hashicorp/terraform-plugin-framework/datasource"
 	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
 // Ensure the implementation satisfies the expected interfaces.
 var (
-	_ datasource.DataSource              = &inventoryDataSource{}
-	_ datasource.DataSourceWithConfigure = &inventoryDataSource{}
+	_ datasource.DataSource              = &InventoryDataSource{}
+	_ datasource.DataSourceWithConfigure = &InventoryDataSource{}
 )
 
 // NewInventoryDataSource is a helper function to simplify the provider implementation.
 func NewInventoryDataSource() datasource.DataSource {
-	return &inventoryDataSource{}
+	return &InventoryDataSource{}
 }
 
 // inventoryDataSource is the data source implementation.
-type inventoryDataSource struct {
+type InventoryDataSource struct {
 	client *AAPClient
 }
 
 // Metadata returns the data source type name.
-func (d *inventoryDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+func (d *InventoryDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
 	resp.TypeName = req.ProviderTypeName + "_inventory"
 }
 
 // Schema defines the schema for the data source.
-func (d *inventoryDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+func (d *InventoryDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
 	resp.Schema = schema.Schema{
 		Attributes: map[string]schema.Attribute{
 			"id": schema.Int64Attribute{
 				Required: true,
 			},
-			"groups": schema.MapNestedAttribute{
-				NestedObject: schema.NestedAttributeObject{
-					Attributes: map[string]schema.Attribute{
-						"hosts": schema.ListAttribute{
-							ElementType: types.StringType,
-							Computed:    true,
-						},
-						"children": schema.ListAttribute{
-							ElementType: types.StringType,
-							Computed:    true,
-						},
-					},
-				},
+			"organization": schema.Int64Attribute{
+				Computed:    true,
+				Description: "Identifier for the organization to which the inventory belongs",
+			},
+			"url": schema.StringAttribute{
+				Computed: true,
+			},
+			"name": schema.StringAttribute{
 				Computed: true,
+				Optional: true,
```

Comment:
I'm just now realizing that all of the attributes except `id` should be computed and _not_ optional, because the user should never be supplying data for those values in the terraform config. If there is no value for the attribute in AAP, the data source instance will just have a null value for that attribute.